game-app/game-core/src/main/java/games/strategy/triplea/ui/PlacementUnitsCollapsiblePanel.java

```diff
@@ -24,7 +25,15 @@ class PlacementUnitsCollapsiblePanel {
     unitsToPlacePanel =
         new SimpleUnitPanel(
             uiContext, SimpleUnitPanel.Style.SMALL_ICONS_WRAPPED_WITH_LABEL_WHEN_EMPTY);
-    panel = new CollapsiblePanel(unitsToPlacePanel, "Placements");
+    panel =
+        new CollapsiblePanel(unitsToPlacePanel, "Placements") {
+          @Override
+          protected void toggleState() {
+            super.toggleState();
+            ClientSetting.placementsCollapsed.setValueAndFlush(isCollapsed());
+          }
```

Comment:
Suggestion to consider:

The override gives me some pause as things could get wonky if the class inheritance were extended further.

Have you considered adding a constructor arg that could do the persistence? I think the key benefit is `toggleState()` could remain encapsulated within `CollapsiblePanel` and there would be no need to 'remember' when override `toggleState()` to do the `super.toggleState()` call.

To explain, the suggestion for consideration would be to add another constructor to `CollapisblePanel`:
```
CollapsiblePanel(panel, title, Consumer<Boolean> toggleStatePersistence) {
```
The existing contructor would be updated to invoke this new one a no-op:
```
CollapsiblePanel(panel, title ) {
   this(panel, title, toggleState -> {});
```
Then the usages would be updated:
```
new CollapisblePanel(unitsToPlacePanel, "Placements",  ClientSetting.placementsCollapsed::setValueAndFlush) 
```

Reply:
Good idea, done!